### PR TITLE
[FIX]using latest good 0.29.35 cython compiler for buildilng; cython=…

### PR DESCRIPTION
**Pull request by marcwimmer**
…=3.0.0 which was recently released does not compile.

message is 

```
     dberrhandle(err_handler)
                  ^
  ------------------------------------------------------------

  src/pymssql/_mssql.pyx:2203:16: Cannot assign type 'int (DBPROCESS *, int, int, int, char *, char *) except? -1' to 'EHANDLEFUNC'

```

I checked the pyx source code and no obvious errors.

During pip3 install the setuptools download the 3.0.0 cython compiler which makes this error. cython==0.29.35 just works fine.

As cython==3.0.0 is released few days ago many people are complaining.

### CODE REVIEW

**Comment by da-woods** [comment]
to make it work on Cython 3 the fix would be to add `noexcept` to the definition of `err_handler`. I suspect pinning the version is a better short -term fix though

**Comment by marcwimmer** [comment]
> 

ah great - i have no experience with python right now. i saw the except? thing - great

**Comment by decaz** [comment]
Why not `~=0.29`? There is 0.29.36 as the latest pre 3 release.

**Comment by marcwimmer** [comment]
whatever - many pipes are broken on azure now and others; needs to get online;
would it match 3.0.29 then also?!? dont know for sure 
but i change it now -- changed!

can you please put it on pypi?

File: pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ requires = [
     "setuptools>=54.0", 
     "setuptools_scm[toml]>=5.0,<7.0", 
     "wheel>=0.36.2", 
-    "Cython>=0.29.22"
+    "Cython==~0.29"
```
Comment (edgarrmondragon) [inline pyproject.toml:6]:
```suggestion
    "Cython~=0.29"
```

Comment (marcwimmer) [reply pyproject.toml:6]: i already changed it to that suggested version - checking if pull request was not updated;
we need that really online.

**Comment by marcwimmer** [comment]
fixed my typo - please online! we need that for a lot of build pipelines that are stuck now.

**Comment by mattjbudd** [comment]
Thank you @marcwimmer for the fix. Hopefully, they can get this online soon!

**Comment by marcwimmer** [comment]
there is an intermediate solution:

use

```
pip3 install cython==0.29.35
pip3 install --no-build-isolation pymssql==2.1.5
```

(at the moment the 2.2.x. packages are broken; 2.1.x works - checked that 2 days ago; could be that they work now)

parameter no build isolation makes, that the current env with cython==0.29.35 is used.

**Comment by yumingchangsabodota** [comment]
when will this be merged and pushed to pip?

**Comment by termim** [comment]
Fixed in #803.

**Pull request closed without merging**